File: 004/problem004.py
#!/usr/bin/env python
def is_palindrome(x):
    return str(x)[::-1] == str(x)


largest_sum = 0

for x in range(0, 1000):
    for y in range(x, 1000):
        product = x * y
        if is_palindrome(product) and product > largest_sum:
            largest_sum = product

print largest_sum

File: 004/problem004.swift
#!/usr/bin/env xcrun swift
extension String {
    var reversed: String {
        return String(self.characters.reversed())
    }
}

extension Int {
    var isPalindrome: Bool {
        return String(self).reversed == String(self)
    }
}

var largest = 0
for x in (100...999) {
    for y in (100...999) {
        let product = x * y
        if product > largest && product.isPalindrome {
            largest = product
        }
    }
}

print(largest)
